webui/ocw/lib/gce.py

```diff
@@ -0,0 +1,65 @@
+from .vault import GCECredential
+import googleapiclient.discovery
+from google.oauth2 import service_account
+from ..models import Instance
+from ..models import ProviderChoice
+from ..models import StateChoice
+from django.db import transaction
+from ..lib import db
+
+
+class GCE:
+    __instance = None
+    __credentials = None
+    __compute_clinet = None
+
+    def __new__(cls):
+        if GCE.__instance is None:
+            GCE.__instance = object.__new__(cls)
+            GCE.__instance.__credentials = GCECredential()
+        return GCE.__instance
+
+    def compute_client(self):
+        if self.__credentials.isExpired():
+            self.__credentials.renew()
+            self.__compute_clinet = None
+        self.project = self.__credentials.getInfo()['project_id']
```

Comment:
why project is not listed in the very beginning of the class and don't have '__' before name ?

Reply:
I think, python don't require it.
But I think it make sense. I will change it.

Reply:
yeah I know , just wanted to understand why you treat project field differently from others ?

Reply:
lazy python programmer :)